### ISSUE TITLE
SharingSettings is null in every new instances of ProvisioningTenant

### ISSUE THREAD

**Issue by orty** [pnp/pnpframework#408]
Hi,

I just bumped into this when trying to implement a `Handler`-filter kind of rule, in which I look into the strongly-typed `ProvisioningTemplate` object for each `Handler` related property, and deciding whether or not the associated `Handler` should be passed to `ApplyProvisioningTemplate(ProvisioningTemplate, ProvisioningTemplateApplyingInformation)`.

ie. for `Files` or `ContentTypes`, I safely check if the collection contains any element. For `Header` or `Footer`, I check if they are null, etc. But for some properties (the ones implementing `IEquatable<>`) I rely on the fact that they are equals (or not) to their default instance equivalent.

Basically, is it true that `myTemplate.Tenant.Equals(new ProvisioningTenant())`. It works like a charm for every properties in the template but the `Tenant` one, because in a new instance of the `ProvisioningTenant` class, the `SharingSettings` property is not initialized (unlike others), and then calls to `.GetHashCode()` or `.Equals()` just throw a `NullReferenceException`.

Would it be a good choice to initialize a `new SharingSettings()` in the property getter ? If yes may I send a pull request to this repository ?

Many thanks !

**Comment by orty** [comment]
Hello again,

Actually, now that I am going further into my development, I see that at least for the `AppCatalog` property (of `ProvisioningTenant` and `ApplicationLifecycleManagement`), the comparison code is relying on a reference comparison (`==`) instead of a value comparison (`.Equals`), which kind of defeats the purpose of implementing `IEquatable<>` in class `AppCatalog`.

Is this behavior the intended one ?
